statsmodels/survey/survey_model.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+
+class SurveyModel(object):
+
+    def __init__(self, design, model_class, init_args={}, fit_args={}):
+        self.design = design
+        self.model = model_class
+        self.init_args = dict(init_args)
+        self.fit_args = dict(fit_args)
+
+    def _centering(self, array=None):
+        if self.center_by == 'est':
+            array -= self.params
+        elif self.center_by == 'global':
+            array -= array.mean(0)
+        elif self.center_by == 'stratum':
+            if self.design.rep_weights is None:
+                for s in range(self.design.nstrat):
+                    # center the 'delete 1' statistic
+                    array[self.design.ii[s], :] -= array[self.design.ii[s],
+                                                         :].mean(0)
+        else:
+            raise ValueError("Centering option not implemented")
+        return array
+
+    def fit(self, y, X, cov_method='jack', center_by='est', replicates=None):
+        self.center_by = center_by
+        self.init_args["weights"] = self.design.weights
+        self.params = self._get_params(y, X)
+        if replicates is None:
```

Comment:
`replicates` sounds like it is a data structure that contains the replicates themselves.  We would usually use `n_reps` or `k_reps` when counting something.